### PR TITLE
Use pattern matching

### PR DESCRIPTION
**Pull request by drewnoakes**
## Proposed changes

- Use C# 9 pattern matching in a few places

## Test methodology 

- Manual testing and review

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommitInfo/CommitInfo.cs [hunk 1]
```diff
@@ -740,8 +739,7 @@ protected override void OnLayout(LayoutEventArgs e)
 
         private void RichTextBox_KeyDown(object sender, KeyEventArgs e)
         {
-            var rtb = sender as RichTextBox;
-            if (rtb is null || !e.Control || e.KeyCode != Keys.C)
+            if (sender is not RichTextBox rtb || !e.Control || e.KeyCode != Keys.C)
```
Comment (gerhardol) [inline GitUI/CommitInfo/CommitInfo.cs:742]: I guess that the pattern matching declaration style is something I have to get used to. In many situations it is probly better too.
I do not like declarations in is not checks though.

**Review by gerhardol — APPROVED** [review]

File: GitUI/CommitInfo/CommitInfo.cs (hunk 1 above)
Comment (drewnoakes) [reply GitUI/CommitInfo/CommitInfo.cs:742]: > I do not like declarations in is not checks though.

It took me a while to get used to it. It's clearer than the alternative, in my opinion, such as:

```c#
if (!(sender is RichTextBox rtb) || ...)
```

Comment (RussKie) [reply GitUI/CommitInfo/CommitInfo.cs:742]: How about this?
```
if (!e.Control || e.KeyCode != Keys.C || sender is not RichTextBox rtb)
```

**Review by RussKie — COMMENTED** [review]
LGTM

File: GitUI/CommitInfo/CommitInfo.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/CommitInfo/CommitInfo.cs:742]: that order is better for my brain, it is kind of the last result that survives the if()

Comment (RussKie) [reply GitUI/CommitInfo/CommitInfo.cs:742]: @drewnoakes can you please reorder (few places), then it is good to merge.
Thanks

File: GitUI/CommandsDialogs/FormPush.cs [hunk 2]
```diff
@@ -1282,10 +1282,10 @@ private void selectTrackedToolStripMenuItem_Click(object sender, EventArgs e)
...
                 {
                     // Check if the branch is tracked (i.e. not new)
-                    var localColumn = row.Cells[LocalColumn.Name] as DataGridViewTextBoxCell;
-                    var remoteColumn = row.Cells[RemoteColumn.Name] as DataGridViewTextBoxCell;
-                    return localColumn is not null && remoteColumn is not null
-                           && !string.IsNullOrEmpty(localColumn.Value.ToString()) && !string.IsNullOrEmpty(remoteColumn.Value.ToString());
+                    return row.Cells[LocalColumn.Name] is DataGridViewTextBoxCell localColumn &&
+                           row.Cells[RemoteColumn.Name] is DataGridViewTextBoxCell remoteColumn &&
+                           !string.IsNullOrEmpty(localColumn.Value.ToString()) &&
+                           !string.IsNullOrEmpty(remoteColumn.Value.ToString());
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormPush.cs:1288]: Left Reading works better with operators at the begin of the next lines.

Comment (drewnoakes) [reply GitUI/CommandsDialogs/FormPush.cs:1288]: Is there an analyzer we could use to control this uniformly across the code?

Comment (RussKie) [reply GitUI/CommandsDialogs/FormPush.cs:1288]: I'm not aware of any. 
@sharwell is the Master of Analyzers, maybe he knows :)

**Pull request merged**